afids_utils/tests/test_io.py

```diff
@@ -0,0 +1,131 @@
+from __future__ import annotations
+
+import csv
+import tempfile
+from os import PathLike, remove
+from pathlib import Path
+
+import numpy as np
+import pytest
+from hypothesis import HealthCheck, assume, given, settings
+from hypothesis import strategies as st
+from numpy.typing import NDArray
+
+from afids_utils.exceptions import InvalidFiducialNumberError
+from afids_utils.io import AFIDS_FIELDNAMES, afids_to_fcsv, get_afid, load_nii
+
+
+@pytest.fixture
+def nii_file() -> PathLike[str]:
+    return (
+        Path(__file__).parent
+        / "data"
+        / "tpl-MNI152NLin2009cAsym_res-01_desc-brain_T1w.nii.gz"
+    )
+
+
+class TestNiftiIO:
+    def test_load_valid_path(self, nii_file: PathLike[str]):
+        nii_affine, nii_data = load_nii(nii_file)
+
+        # Check output types
+        assert isinstance(nii_affine, np.ndarray)
+        assert isinstance(nii_data, np.ndarray)
+        # Check element types
+        assert nii_affine.dtype == np.single
+        assert nii_data.dtype == np.single
+
+
+@pytest.fixture
+def valid_fcsv_file() -> PathLike[str]:
+    return (
+        Path(__file__).parent / "data" / "tpl-MNI152NLin2009cAsym_afids.fcsv"
+    )
+
+
+class TestGetAfid:
+    @given(afid_num=st.integers(min_value=1, max_value=32))
+    @settings(
+        suppress_health_check=[HealthCheck.function_scoped_fixture],
+    )
+    def test_valid_num_get_afid(
+        self, valid_fcsv_file: PathLike[str], afid_num: int
+    ):
+        afid = get_afid(valid_fcsv_file, afid_num)
+
+        # Check array type
+        assert isinstance(afid, np.ndarray)
+        # Check array values
+        assert afid.dtype == np.single
+
+    @given(afid_num=st.integers(min_value=-1000, max_value=1000))
+    @settings(
+        suppress_health_check=[HealthCheck.function_scoped_fixture],
+    )
+    def test_invalid_num_get_afid(
+        self,
+        valid_fcsv_file: PathLike[str],
+        afid_num: int,
+    ):
+        assume(afid_num < 1 or afid_num > 32)
+
+        with pytest.raises(
+            InvalidFiducialNumberError, match=".*is not valid."
+        ):
+            _ = get_afid(valid_fcsv_file, afid_num)
+
+
+@pytest.fixture
+def afids_coords() -> NDArray[np.single]:
```

Comment:
There is `hypothesis.extra.numpy.arrays` for this

Reply:
Yeah, I will fix this - I learned about the numpy.arrays after initially implementing this.